sherpa/bin/pruned_stateless_emformer_rnnt2/streaming_server.py

```diff
@@ -131,11 +136,45 @@ def get_args():
         """,
     )
 
+    parser.add_argument(
+        "--beam",
+        type=float,
+        default=10.0,
+        help="""A floating point value to calculate the cutoff score during beam
+            search (i.e., `cutoff = max-score - beam`), which is the same as the
+            `beam` in Kaldi.
+            Used only when --decoding-method is fast_beam_search.
+            """,
+    )
+
+    parser.add_argument(
+        "--max-contexts",
+        type=int,
+        default=8,
+        help="""Used only when --decoding-method is fast_beam_search.""",
+    )
+
+    parser.add_argument(
+        "--max-states",
+        type=int,
+        default=32,
+        help="""Used only when --decoding-method is fast_beam_search.""",
+    )
+
+    parser.add_argument(
+        "--decoding-method",
+        type=str,
+        default="greedy_search",
+        help="""Decoding method to use. Currently, only greedy_search and
+            fast_beam_search are implemented.
+            """,
+    )
+
     return parser.parse_args()
 
 
 @torch.no_grad()
-def run_model_and_do_greedy_search(
+def run_model_and_do_search(
```

Comment:
I suggest refactoring it into two functions:
- run_model_and_do_greedy_search
- run_model_and_do_fast_beam_search

It makes the code more readable and more maintainable, I think.

If we want to support another decoding method, we can just create a new function to do that.

----

It is just a suggestion. You can leave it as it is.

Reply:
It make sense, I will finish first the work of conv_emformer at first